frontend/common/src/components/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,123 @@
+import React from "react";
+import { XIcon } from "@heroicons/react/outline";
+
+import Overlay from "./Overlay";
+import Content from "./Content";
+
+import "@reach/dialog/styles.css";
+import "./dialog.css";
+
+export type Color = "primary" | "secondary" | "tertiary";
```

Comment:
How will this work if we share the component between talentsearch and indigenousapprenticeship?  When I set up the hello world PR I added the *ia* prefix to the colours in the hydrogen config.  Maybe something like that?

Reply:
I can add the ia prefix to match that, since the header has a gradient background i simply used the three gradients available in the hydrogen config.

We could make it more explicit and do `type Color = "purple" | "ia-purple" | "ia-pink";`?

Reply:
Yeah, I like that better.  I think the best would be something like "ts-primary" and "ia-primary" but I'm not sure the designers have designated colours like that.

Reply:
Updated with [update colour names](https://github.com/GCTC-NTGC/gc-digital-talent/pull/2561/commits/36b7ae19914a86f04a5c627dbb4faece63a6b0de). I used your suggestion of prefixing primary, secondary etc. to match the button component more closely.